python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -0,0 +1,308 @@
+    ...                    'month': [2, 3],
+    ...                    'day': [4, 5]})
+    >>> cudf.to_datetime(df)
+    0   2015-02-04
+    1   2016-03-05
+    dtype: datetime64[ns]
+    >>> cudf.to_datetime(1490195805, unit='s')
+    numpy.datetime64('2017-03-22T15:16:45.000000000')
+    >>> cudf.to_datetime(1490195805433502912, unit='ns')
+    numpy.datetime64('1780-11-20T01:02:30.494253056')
+    """
+    if arg is None:
+        return None
+
+    if exact is False:
+        raise NotImplementedError("exact support is not yet implemented")
+
+    if origin != "unix":
+        raise NotImplementedError("origin support is not yet implemented")
+
+    if yearfirst:
+        raise NotImplementedError("yearfirst support is not yet implemented")
+
+    try:
+        if isinstance(arg, cudf.DataFrame):
+            # we require at least Ymd
+            required = ["year", "month", "day"]
+            req = list(set(required) - set(arg._data.names))
+            if len(req):
+                req = ",".join(req)
+                raise ValueError(
+                    "to assemble mappings requires at least that "
+                    f"[year, month, day] be specified: [{req}] "
+                    "is missing"
+                )
+
+            # replace passed column name with values in _unit_map
+            unit = {k: get_units(k) for k in arg._data.names}
+            unit_rev = {v: k for k, v in unit.items()}
+
+            # keys we don't recognize
+            excess = set(unit_rev.keys()) - set(_unit_map.values())
+            if len(excess):
+                excess = ",".join(excess)
+                raise ValueError(
+                    f"extra keys have been passed to the \
+                        datetime assemblage: [{excess}]"
+                )
+
+            new_series = (
+                arg[unit_rev["year"]].astype("str")
+                + "-"
+                + arg[unit_rev["month"]].astype("str").str.zfill(2)
+                + "-"
+                + arg[unit_rev["day"]].astype("str").str.zfill(2)
+            )
+            format = "%Y-%m-%d"
+            col = new_series._column.as_datetime_column(
+                "datetime64[s]", format=format
+            )
+
+            for u in ["h", "m", "s", "ms", "us", "ns"]:
+                value = unit_rev.get(u)
+                if value is not None and value in arg:
+                    arg_col = arg._data[value]
+                    if arg_col.dtype.kind in ("f"):
+                        col = new_series._column.as_datetime_column(
+                            "datetime64[ns]", format=format
+                        )
+                        break
+                    elif arg_col.dtype.kind in ("O"):
+                        if not cpp_is_integer(arg_col).all():
+                            col = new_series._column.as_datetime_column(
+                                "datetime64[ns]", format=format
+                            )
+                            break
```

Comment:
In this case, not really - Because here were are actually deciding if any of `["h", "m", "s", "ms", "us", "ns"]` are floating values in form of strings, we will type-cast to `datetime64[ns]`. Else we are good with the existing dtype of `col`.

Reply:
So these are strings of floats as opposed to strings of datetime-like values? I.E. "1997-02-05 10:05:37.663"

Reply:
Yep correct. Something like this..
```python
>>> df = pd.DataFrame({'year': [2015, 2016],
                   'month': [2, 3],
                   'day': [4, 5],
                    'hour': ["6.43545", "0.2323"]})

>>>  pd.to_datetime(df)
0   2015-02-04 06:26:07.620
1   2016-03-05 00:13:56.280
dtype: datetime64[ns]
```

Reply:
Ah okay, that makes a lot more sense, thanks!